src/styles/blocks/mains.scss

```diff
@@ -0,0 +1,50 @@
+.calendar {
+  position: relative;
+  display: flex;
+  flex-wrap: wrap;
+
+  padding: 25px $calendarShihtPading 10px ;
+  gap: $gapForDays;
+  margin: auto;
+
+  width: $calendarMaxWidth;
+
+  @each $day in $days {
+    &--start-day-#{$day} > .calendar__day:first-child {
+      margin-left: $startDayShift;
+    }
+  }
```

Comment:
it'll not working as task expects. It should set margin-left depends from value of start-day-{day}.
Now it's has static gap when calendar starts from sunday.
you should have something like 
$days:
  "mon" 1,
  "tue" 2,
  "wed" 3,
  "thu" 4,
  "fri" 5,
  "sat" 6,
  "sun" 7;
  
 depends from it you calculate margin.